relay-event-normalization/src/normalize/span/description/sql/parser.rs

```diff
@@ -235,9 +231,14 @@ impl VisitorMut for NormalizeVisitor {
     }
 
     fn post_visit_expr(&mut self, expr: &mut Expr) -> ControlFlow<Self::Break> {
-        if let Expr::CompoundIdentifier(parts) = expr {
-            Self::simplify_compound_identifier(parts);
+        // Casts are omitted for simplification. Because we replace the entire expression,
+        // the replacement has to occur *after* visiting its children.
+        if let Expr::Cast { expr: inner, .. } = expr {
+            let mut swapped = Expr::Value(Value::Null);
+            std::mem::swap(&mut swapped, inner);
+            *expr = swapped;
```

Comment:
This is clumsy, but I wanted to prevent cloning `inner`. Open to better ideas.

Reply:
It seems expressions are generally updating in the pre_visit. Is there any reason to do this one in the post_visit?

Reply:
Yep, the reasoning is in lines 234 - 235! When I replace `expr` in the pre-visit, it's children are not being visited anymore.